### ISSUE TITLE
Reseting the date of incorporation crashes the app 

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14417]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
 1.Connect a bank account
 2. Under company information, select incorporation date
 3. Click on Reset option

## Expected Result:
The date should be reset to today

## Actual Result:
App crashes

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] iOS / native

**Version Number:** 1.2.56-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Could not check
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/213555480-f292e96a-fd04-4e9d-aa68-ec79d8215e00.mov

https://user-images.githubusercontent.com/43996225/213555615-5d5c8021-76f1-4c4b-b9ac-0c104c91146b.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @esh-g 
**Slack conversation:**  https://expensify.slack.com/archives/C049HHMV9SM/p1674144058692939

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by marcochavezf** [comment]
I reproduced the bug locally, seems we didn't implement an initial value for the `DatePicker` on iOS. Putting a PR shortly.

**Comment by marcochavezf** [comment]
[PR](https://github.com/Expensify/App/pull/14427) up.

**Comment by marcochavezf** [comment]
Oh noticed after creating the PR that @esh-g proposed a fix [here](https://github.com/Expensify/App/pull/14270#issuecomment-1397423655). Not sure if we should compensate since the solution is straightforward.

**Comment by mallenexpensify** [comment]
>Oh noticed after creating the PR that @esh-g proposed a fix https://github.com/Expensify/App/pull/14270#issuecomment-1397423655. Not sure if we should compensate since the solution is straightforward.

@marcochavezf , can you elaborate on this please?  My understand is that there was a regression and @esh-g caught it, then we fixed it.  If so, they're, at least, due compensation for finding the regression and reporting it.   If they also provided the solution that we used, and you're wondering if compensation is due for that, we'd need to discuss more (cuz I'm unsure)

**Comment by esh-g** [comment]
> >Oh noticed after creating the PR that @esh-g proposed a fix https://github.com/Expensify/App/pull/14270#issuecomment-1397423655. Not sure if we should compensate since the solution is straightforward.
> 
> @marcochavezf , can you elaborate on this please?  My understand is that there was a regression and @esh-g caught it, then we fixed it.  If so, they're, at least, due compensation for finding the regression and reporting it.   If they also provided the solution that we used, and you're wondering if compensation is due for that, we'd need to discuss more (cuz I'm unsure)

Yes, I thought this would work like reporting a regular bug as this PR was merged into staging....
And it took some work finding the PR it was linked two, at first I incorrectly commented on two PRs before finding the right one 😅
And I guess, my solution could be the proposal for this bug

**Comment by marcochavezf** [comment]
>@marcochavezf , can you elaborate on this please? My understand is that there was a regression and @esh-g caught it, then we fixed it. If so, they're, at least, due compensation for finding the regression and reporting it. 

I agree we should compensate for finding and reporting it.

>If they also provided the solution that we used, and you're wondering if compensation is due for that, we'd need to discuss more (cuz I'm unsure)

Yeah, I saw the solution after I created the PR since it's a one-line change, so technically I didn't use the solution provided but it's the same solution 😄

**Comment by mallenexpensify** [comment]
Thanks @marcochavezf , i'll plan to compensate @esh-g $250 for reporting the regression when I pay out this issue. 

@aimane-chnaif I think the issue for assignment got borked cuz the PR was created at the same time as you were assigned here.  Let's consider it an edge case for now unless anyone disagrees

**Comment by esh-g** [comment]
The PR #14427 is merged, so is this issue now resolved?

**Comment by esh-g** [comment]
Yes, just wanted to confirm that it was on the 7-day period 👍

**Comment by mallenexpensify** [comment]
Thanks @thesahindia , sorry for the delay @esh-g , the issue wasn't having the `Overdue` label added cuz of the `Reviewing` label. Can you please apply and comment here once you have?
https://www.upwork.com/jobs/~015f5204e26a9f2c93

It might say $1000 for the job, payment will be $250 for reporting.

**Comment by esh-g** [comment]
@mallenexpensify  I have applied under the name Tanya G.

**Comment by mallenexpensify** [comment]
@esh-g and @thesahindia both hired, can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~015f5204e26a9f2c93

**Comment by esh-g** [comment]
@mallenexpensify Accepted, successfully!

**Comment by mallenexpensify** [comment]
@esh-g paid $250 for reporting
@thesahindia paid $1000 for C+ PR review. 

@thesahindia can you complete steps 1-3 below? 
 
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] The PR that introduced the bug has been identified. Link to the PR:
- [ ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here: 

Checking on this the last step here https://expensify.slack.com/archives/C01SKUP7QR0/p1675906686840469 

Regression Test Buddy Check!
Context: [Reseting the date of incorporation crashes the app #14417](https://github.com/Expensify/App/issues/14417)
Recommendation: Add step 5 and 6 under "Online behavior (Continue with bank account flow)" to https://expensify.testrail.io/index.php?/cases/view/1971143  
5. Click reset 
6. Verify The date is reset to today and that the app doesn't crash. 
 Please :yes: or :no: if you think this is the right path forward! Thank you!
